internal/host/validator.go

```diff
@@ -658,4 +659,157 @@ func (v *validator) printSufficientOrUnknownInstallationDiskSpeed(c *validationC
 	default:
 		return fmt.Sprintf("Unexpected status %s", status.String())
 	}
+
+}
+
+func (v *validator) hasSufficientNetworkLatencyRequirementForRole(c *validationContext) ValidationStatus {
+
+	if len(c.cluster.Hosts) == 1 {
+		// Single Node use case
+		return ValidationSuccess
+	}
+	if len(c.host.Connectivity) == 0 {
+		return ValidationPending
+	}
+	latency, err := v.validateNetworkLatencyForRole(c.host, c.clusterHostRequirements, c.cluster.Hosts)
+	if err != nil || len(latency) > 0 {
+		return ValidationFailure
+	}
+	return ValidationSuccess
+}
+
+func getHostnameAndRoleByIP(ip string, hosts []*models.Host) (string, models.HostRole, error) {
+
+	for _, h := range hosts {
+		if h.Inventory != "" {
```

Comment:
```suggestion
		if h.Inventory == "" {
		    continue
		}
		
		inv, err := hostutil.UnmarshalInventory(h.Inventory)
	        if err != nil {
		        return "", "", fmt.Errorf("unable to unmarshall cluster inventory for host %s", h.RequestedHostname)
	        }
	        
	        if funk.Find(inv.Interfaces, func(i *models.Interface) bool {
	            return funk.Find(i.IPV4Addresses, func(ipv4 string) bool { ipv4 == ip} || funk.Find(i.IPV6Addresses, func(ipv6 string) bool { ipv6 == ip}
	        } {
	            return getRealHostname(h, inv), h.Role, nil
	        }
	        
		
```

Reply:
I'll add your suggestion in my next update. Thanks!